src/resources/nfts/index.ts

```diff
@@ -40,21 +40,21 @@ export function useNFTs(): NFT[] {
 export function useLegacyNFTs({ address }: { address: string }) {
   const { accountAddress } = useAccountSettings();
   const isOwner = accountAddress === address;
-  const { data: polygonAllowlist } = usePolygonAllowlist();
   const {
     data,
     error,
     fetchNextPage,
     hasNextPage,
-    isFetching,
     isFetchingNextPage,
+    isFetching,
   } = useInfiniteQuery({
     queryKey: nftsQueryKey({ address }),
     queryFn: async ({ pageParam }) => {
-      const { data, nextCursor } = await fetchSimpleHashNFTs(
-        address,
-        pageParam
-      );
+      const [simplehashResponse, polygonAllowlist] = await Promise.all([
```

Comment:
the previous iteration of this hook was somehow remaining `enabled` even when the polygonAllowlist did not exist, so fixed it by fetching it every time it's needed. performance-wise there is no difference bc the polygon allowlist query has a 10 min stale time, which means 99% of the time the polygon allowlist will return immediately without refetching.